Consul.Test/CatalogTest.cs

```diff
@@ -225,19 +225,25 @@ public async Task Catalog_GetNodesForMeshCapableService()
                 Port = 8000,
                 Connect = new AgentServiceConnect
                 {
-
-                    SidecarService = new AgentServiceRegistration
-                    {
-                        Name = "sidecar",
-                        Port = 9000,
-                    },
+                    SidecarService = new AgentServiceRegistration { Name = "sidecar", Port = 9000, },
                 },
             };
-            await _client.Agent.ServiceRegister(registration);
 
+            await _client.Agent.ServiceRegister(registration);
+            // First request - no WaitIndex set
             var services = await _client.Catalog.NodesForMeshCapableService(registration.Name);
+            if (services.Response == null || services.Response.Length == 0)
+            {
+                //Assert.NotEmpty(services.Response);
+                //Assert.Equal(services.Response[0].ServiceID, registration.Name + "-sidecar-proxy");
+                // Get the index from the first response
+                var currentIndex = services.LastIndex;
```

Comment:
No need to instantiate  a new 'currentIndex' variable here. It doesn't add any value.